server/src/main/java/org/opensearch/monitor/fs/FsInfo.java

```diff
@@ -75,16 +75,27 @@ public static class Path implements Writeable, ToXContentObject {
         long free = -1;
         long available = -1;
         long fileCacheReserved = -1;
-        long fileCacheUtilized = 0;
+        long fileCacheUtilized = -1;
 
         public Path() {}
 
-        public Path(String path, @Nullable String mount, long total, long free, long available) {
+        // Used only in tests
```

Comment:
Please make it package private instead of adding comments

Reply:
@reta Unfortunately, this can not be made package private without large refactoring. It is used in several tests across various packages.

I will be removing many comments that I left in this PR in my future update, I used them to give a small hint for a reviewers in order to explain some of my intentions and decisions. In general I am not a fan of growing list of method parameters (IMO it can be a sign of a weak API design) but in this case I was willing to go this direction because it is "just" a testing method.

One option was to introduce a second constructor with more parameters and keep the old constructor with less parameters and silently use some "default" values for the new parameters inside (in this case `zeroes`). This way we would not have to modify some testing classes. However; I have decided not to do it for a very specific reason and that is: the code that introduced file cache into Opensearch took me some time to figure out, at least for me it was not really easy to understand and I believe using some magic default values would not help the readability of testing code.

Reply:
I just realized that this is `@PublicApi(since = "1.0.0")`, that would be a breaking change. So please please keep the old constructor and introduce the new one, if possible as package internal,

Reply:
Technically speaking, this PR changes the output of `toXContent` method of `FsInfo.Path` with is also `@PublicApi(since = "1.0.0")`. Is that also considered a breaking change? (Http clients are impacted)

Reply:
> This in an API change (fix) that needs documentation update

In fact, this whole PR was born because file caches are currently missing in nodes stats documentation. I was about to send PR into documentation but I realized the code was strange and contained bugs. So fixing code first, documentation follows up. (There are still a few other missing metrics in nodes stats besides file caches, all on my list...)

Reply:
> So please please keep the old constructor and introduce the new one, if possible as package internal,

Unfortunately, can't be package private. It's used across several (test) packages. I will create a second public ctor with more parameters then.

I'm also considering deprecating the "old" ctor because it makes no sense to keep it around (it won't be used in this code) - however, this would mean we would have to maintain it for a while and remove it with some future major release and I am aware this could be viewed as a controversial change.

I don't know. It feels like we've created an "API life-cycle" issue that isn't helpful in situation like this. If I'm not mistaken it's a frequent pattern in OpenSearch code base to have dedicated ctors for certain type of objects just for the tests (and they should never be really used anywhere else). Either we should have excluded these ctors from public "API life-cycle" (adding special type of annotation like `@testCope` sounds ugly to me but could work) or we need to acknowledge that we need to carry on some API maintenance gymnastics going forward. 

IMO the clean solution would be to change the signature of the ctor (again, which is meant to be used in tests only). There is a reason for this because since the file caches were introduced then the `FsInfo.Path` simply should signal that file cache variables are important part of it. And anyone who has been using the existing ctor of the Path should be forced to review his/her code and adapt properly.

Anyway, I am happy to implement whatever is recommended as part of PR review but I wanted to clearly explain my position.

Reply:
> Unfortunately, can't be package private. It's used across several (test) packages (for example see `NodeStatsTests` for serialization tests). I will create a second public ctor with more parameters then.

👍 

> Either we should have excluded these ctors from public "API life-cycle" (adding special type of annotation like @testCope sounds ugly to me but could work) or we need to acknowledge that we need to carry on some API maintenance gymnastics going forward.

User `@InternalApi` for constructor

Reply:
> Use @InternalApi for constructor

👍 I will use it for the new ctor but please note that IMO it is perfectly fine if this ctor is used by any plugin author (so the _"... should be not used outside of OpenSearch core components"_ does not 100% fit here).